File: app_core/core/src/main/java/com/wangzs/core/network/interceptor/RxLoggingInterceptor.java
package com.wangzs.core.network.interceptor;


import com.wangzs.base.toolskit.LogUtil;

import org.jetbrains.annotations.NotNull;

import okhttp3.logging.HttpLoggingInterceptor;

/**
 * @Description:
 * @Author: wangzs
 * @Date: 2022-03-10
 * @Version:
 */
public class RxLoggingInterceptor {

    /**
     * 输出log
     *
     * @return
     */
    public static HttpLoggingInterceptor create() {
        return new HttpLoggingInterceptor(new HttpLoggingInterceptor.Logger() {
            @Override
            public void log(@NotNull String message) {
                LogUtil.i(message);
            }
        }).setLevel(HttpLoggingInterceptor.Level.BODY);
    }
}

File: app_core/core/src/main/java/com/wangzs/core/network/interceptor/RxLoggingInterceptor.kt
package com.wangzs.core.network.interceptor

import com.wangzs.base.toolskit.LogUtil
import okhttp3.logging.HttpLoggingInterceptor

/**
 * @Description:
 * @Author: wangzs
 * @Date: 2020-03-10
 * @Version:
 */
object RxLoggingInterceptor {
    /**
     * 输出log
     *
     * @return
     */
//    fun create(): HttpLoggingInterceptor {
//        return HttpLoggingInterceptor(object : HttpLoggingInterceptor.Logger {
//            override fun log(message: String) {
//                LogUtil.i(message)
//            }
//        }).setLevel(HttpLoggingInterceptor.Level.BODY)
//    }
    fun create(): HttpLoggingInterceptor {
        return HttpLoggingInterceptor().apply {
            level = HttpLoggingInterceptor.Level.BODY
        }
    }

}